src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -942,40 +1004,67 @@ static int rbd_nbd(int argc, const char *argv[])
   }
 
   if (args.begin() != args.end()) {
-    cerr << "rbd-nbd: unknown args: " << *args.begin() << std::endl;
-    return EXIT_FAILURE;
+    *err_msg << "rbd-nbd: unknown args: " << *args.begin();
+    return -EINVAL;
+  }
+
+  return 0;
+}
+
+static int rbd_nbd(int argc, const char *argv[])
+{
+  int r;
+
+  Config cfg;
+
+  vector<const char*> args;
```

Comment:
This looks like too many unnecessary empty lines too me. I think we can group these declarations in one section (not separating by empty lines).